From c685bc58489a74c54991476f5da17a1864030d06 Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Fri, 17 Jul 2026 11:15:25 +0200
Subject: [PATCH] doc: libnftables-json: Describe RULESET object

JIRA: https://issues.redhat.com/browse/RHEL-190549
Upstream Status: nftables commit 454f361434522bbeba32e114a14c336e1ebf20a1

commit 454f361434522bbeba32e114a14c336e1ebf20a1
Author: Phil Sutter <phil@nwl.cc>
Date:   Thu Nov 6 12:14:56 2025 +0100

    doc: libnftables-json: Describe RULESET object

    Document the syntax of this meta-object used by "list" and "flush"
    commands only.

    Fixes: 872f373dc50f7 ("doc: Add JSON schema documentation")
    Signed-off-by: Phil Sutter <phil@nwl.cc>

Signed-off-by: Phil Sutter <psutter@redhat.com>
---
 doc/libnftables-json.adoc | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/doc/libnftables-json.adoc b/doc/libnftables-json.adoc
index 9129cc8..259c1e3 100644
--- a/doc/libnftables-json.adoc
+++ b/doc/libnftables-json.adoc
@@ -200,6 +200,18 @@ Rename a chain. The new name is expected in a dedicated property named
 
 == RULESET ELEMENTS
 
+=== RULESET
+[verse]
+____
+*{ "ruleset":* 'RULSET_PROPERTIES' *}*
+
+'RULESET_PROPERTIES' := *null* | *{ "family":* 'STRING' *}*
+____
+
+This is a special object for use with *list* and *flush* commands which will
+then operate on either the whole ruleset or the parts of it belonging to the
+given family.
+
 === TABLE
 [verse]
 ____
